packages/engine/lib/execution/src/worker/mod.rs

```diff
@@ -11,4 +30,809 @@ pub use self::{
         ContextBatchSync, StateSync, SyncCompletionReceiver, SyncCompletionSender, SyncPayload,
         WaitableStateSync,
     },
+    task::{WorkerTask, WorkerTaskResultOrCancelled},
+};
+use crate::{
+    runner::{
+        comms::{
+            ExperimentInitRunnerMsg, InboundToRunnerMsgPayload, NewSimulationRun,
+            OutboundFromRunnerMsg, OutboundFromRunnerMsgPayload, RunnerTaskMessage,
+        },
+        JavaScriptRunner, Language, MessageTarget, PythonRunner, RunnerConfig, RustRunner,
+    },
+    task::{SharedState, TaskId, TaskMessage, TaskResultOrCancelled, TaskSharedStore},
+    worker::pending::{CancelState, PendingGroup, PendingWorkerTask, PendingWorkerTasks},
+    worker_pool::comms::{
+        WorkerCommsWithWorkerPool, WorkerPoolToWorkerMsg, WorkerPoolToWorkerMsgPayload,
+        WorkerToWorkerPoolMsg,
+    },
+    Error, Result,
 };
+
+/// A task worker.
+///
+/// Represents three dedicated language workers.
```

Comment:
Do you want me to add this to the docs as well? A `Worker` is created with `RunnerSpawnConfig`, so the caller is already aware, that those runners could be disabled. It might makes sense to add this to the `Worker` docs for clarification.

Reply:
```rust
/// A task worker containing three dedicated language runners.
/// 
/// Depending on the [`RunnerSpawnConfig`] provided to the `Worker`, different language runners may
/// be enabled or disabled.
```

Reply:
If we go with them being "disabled" but the worker still contains them I'm fine with it yeah :D